packages/web3-eth/test/integration/defaults.test.ts

```diff
@@ -0,0 +1,525 @@
+			const balance = await eth2.getBalance(acc.address);
+			expect(balance).toBe('0x0');
+			eth2.setConfig({
+				defaultBlock: 'latest',
+			});
+			const balanceLatest = await eth2.getBalance(acc.address);
+			expect(Number(hexToNumber(balanceLatest))).toBeGreaterThan(0);
+		});
+		it('transactionBlockTimeout', () => {
+			// default
+			expect(web3Eth.transactionBlockTimeout).toBe(50);
+
+			// after set
+			web3Eth.setConfig({
+				transactionBlockTimeout: 1,
+			});
+			expect(web3Eth.transactionBlockTimeout).toBe(1);
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					transactionBlockTimeout: 120,
+				},
+			});
+			expect(eth2.transactionBlockTimeout).toBe(120);
+		});
+		it('transactionConfirmationBlocks', () => {
+			// default
+			expect(web3Eth.transactionConfirmationBlocks).toBe(24);
+
+			// after set
+			web3Eth.setConfig({
+				transactionConfirmationBlocks: 3,
+			});
+			expect(web3Eth.transactionConfirmationBlocks).toBe(3);
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					transactionConfirmationBlocks: 4,
+				},
+			});
+			expect(eth2.transactionConfirmationBlocks).toBe(4);
+			// implementation tested here ./watch_transaction.test.ts
+		});
+		it('transactionPollingInterval and transactionPollingTimeout', () => {
+			// default
+			expect(web3Eth.transactionPollingInterval).toBe(1000);
+			expect(web3Eth.transactionPollingTimeout).toBe(750);
+
+			// after set
+			web3Eth.setConfig({
+				transactionPollingInterval: 3,
+				transactionPollingTimeout: 10,
+			});
+			expect(web3Eth.transactionPollingInterval).toBe(3);
+			expect(web3Eth.transactionPollingTimeout).toBe(10);
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					transactionPollingInterval: 400,
+					transactionPollingTimeout: 10,
+				},
+			});
+			expect(eth2.transactionPollingInterval).toBe(400);
+			expect(eth2.transactionPollingTimeout).toBe(10);
+		});
+		it('transactionReceiptPollingInterval and transactionConfirmationPollingInterval', () => {
+			// default
+			expect(web3Eth.transactionReceiptPollingInterval).toBeNull();
+			expect(web3Eth.transactionConfirmationPollingInterval).toBeNull();
+
+			// after set
+			web3Eth.setConfig({
+				transactionReceiptPollingInterval: 3,
+				transactionConfirmationPollingInterval: 10,
+			});
+			expect(web3Eth.transactionReceiptPollingInterval).toBe(3);
+			expect(web3Eth.transactionConfirmationPollingInterval).toBe(10);
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					transactionReceiptPollingInterval: 400,
+					transactionConfirmationPollingInterval: 10,
+				},
+			});
+			expect(eth2.transactionReceiptPollingInterval).toBe(400);
+			expect(eth2.transactionConfirmationPollingInterval).toBe(10);
+		});
+		it('blockHeaderTimeout', () => {
```

Comment:
In future we should have test for `blockHeaderTimeout` with actual tx's receipt wait on socket provider

Reply:
https://github.com/ChainSafe/web3.js/issues/5112